_tasks/sync-proposal-data.mjs

```diff
@@ -0,0 +1,346 @@
+    : yaml.dump(src, options);
+  return Buffer.from(ymlDocument);
+}
+
+/**
+ * Converts the presentation date Markdown link to the data structure object.
+ * @param {string} dateUrlTxtVec
+ * @returns {PresentedObj}
+ */
+function toPresentedObj(dateUrlTxtVec) {
+  // Ex: <sub>[December&#xA0;2019][nonblocking-notes]</sub>
+  dateUrlTxtVec = dateUrlTxtVec.slice(5, -5); // Remove wrapping <sub>s.
+  return {
+    date: getTxtVecContent(dateUrlTxtVec, 0),
+    url: getUrlFromDoc(getTxtVecContent(dateUrlTxtVec, 1)),
+  };
+}
+
+// -----------------------------------------------------------------------------
+// Main
+// -----------------------------------------------------------------------------
+
+const tblRegex = new RegExp(/(### Stage 3\n\n)([.+?\s\S]+)(\n\n### Stage 2)/g);
+const results = tblRegex.exec(proposalsReadmeMarkdown);
+
+// Our second capture group: Markdown string with a column-oriented table.
+const markdownTable = results[2];
+
+// It would be preferable to convert the table directly from Markdown -> JSON,
+// however, there does not appear to be any quality packages that offer this.
+// Therefore, we will Markdown -> HTML -> JSON instead.
+// -----------------------------------------------------------------------------
+
+// Markdown -> HTML
+
+let tableHtml;
+
+unified()
+  .use(markdown)
+  .use(html)
+  .process(markdownTable, function (err, file) {
+    ifError(err);
+    tableHtml = String(file);
+  });
+
+// HTML -> JSON
+
+const tablesAsJson = table2json.convert(tableHtml, {
+  stripHtmlFromCells: false,
+});
+const firstTableAsJson = tablesAsJson[0];
+
+// With the table now as JSON, all that's left to do is to augment the fields to
+// achieve the data structure's final form before the JSON -> YAML conversion.
+// -----------------------------------------------------------------------------
+
+// First thing's first, we need to make some adjustments to our data structure.
+
+const dataStruct = firstTableAsJson.map(async function (value) {
+  // ! Some members of this hashmap are only present under certain conditions.
+  const hashmap = new Map();
+
+  const /** @type {string} */ prpslTitle = getTxtVecContent(value.Proposal, 0);
+  hashmap.set('title', prpslTitle);
+
+  const /** @type {string} */ urlKeyLike = getTxtVecContent(value.Proposal, 1);
+  const /** @type {string} */ prpslId = getPrpslId(urlKeyLike);
+  hashmap.set('id', prpslId);
+
+  const /** @type {string} */ codeSample = await getCodeSample(prpslId);
+  if (codeSample.length > 0) {
+    hashmap.set('example', codeSample.pop());
+  }
+
+  const /** @type {Array<PresentedObj>} */ presentedObjArr = [
+    toPresentedObj(value['Last Presented']),
+  ];
+  hashmap.set('presented', presentedObjArr);
+
+  hashmap.set('has_specification', await hasSpec(prpslId));
+
+  // TODO(DerekNonGeneric): Determine where to reliably locate descriptions.
+  const /** @type {string} */ description = 'Lorem ipsum.';
+  hashmap.set('description', description);
```

Comment:
I was thinking that using the description of the GitHub repo would be acceptable, but that doesn't seem to be the currently followed practice. It's possible that these one-line descriptions are all located in the same place for each proposal and I haven't noticed where to look yet. To illustrate my point, take `proposal-static-class-features` for example.

Current proposal description in `stage3.yml`:

> This proposal adds Static public fields, Static private methods and Static private fields

Current proposal description in GitHub repo UI:

> The static parts of new class features, in a separate proposal

---

Currently, parsing each proposal's README for a one-line description would be complicated since these descriptions seem to be derived from non-standard locations in the document, so that seems like a no-go.